components/ee/agent-smith/pkg/agent/agent.go

```diff
@@ -211,12 +213,29 @@ func (agent *Smith) Start(ctx context.Context, callback func(InfringingWorkspace
 		clo = make(chan classifiedProcess, 50)
 	)
 	agent.metrics.RegisterClassificationQueues(cli, clo)
+
+	workspaces := make(map[int]*common.Workspace)
```

Comment:
I did not add it as part of AgentSmith struct as the struct currently does not maintain any kind of state.